IdentityCore/src/JWEResponse/MSIDJweResponse+EcdhAesGcm.m

```diff
@@ -0,0 +1,251 @@
+                                                              error:(NSError * _Nullable __autoreleasing * _Nullable)error
+{
+    SecKeyAlgorithm algorithm = kSecKeyAlgorithmECDHKeyExchangeStandard;
+    NSDictionary *attributes = @{(id)kSecKeyKeyExchangeParameterRequestedSize:@32};
+    
+    CFErrorRef cfError = NULL;
+    NSData *sharedKey = (NSData *)CFBridgingRelease(SecKeyCopyKeyExchangeResult(privateStkRef, algorithm, serverKeyRef, (__bridge CFDictionaryRef)attributes, &cfError));
+    
+    if (!sharedKey)
+    {
+        NSError *nsError = CFBridgingRelease(cfError);
+        
+        if (error)
+        {
+            *error = nsError;
+        }
+        
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to do key exchange with error %@", nsError);
+        return nil;
+    }
+    
+    return sharedKey;
+}
+
+- (NSData *)calculateDerivedKeyWithSharedKey:(NSData *)sharedSecret
+                                 algorithmId:(NSData *)algorithmId
+                                         apu:(NSData *)apu
+                                         apv:(NSData *)apv
+                                       error:(NSError * _Nullable __autoreleasing * _Nullable)error
+{
+    NSError *concatKDFError = nil;
+    MSIDConcatKdfProvider *concatKDFProvider = [MSIDConcatKdfProvider new];
+    NSData *derivedKey = [concatKDFProvider concatKDFWithSHA256WithSharedSecret:sharedSecret
+                                                           outputKeyLen:32
+                                                            algorithmId:algorithmId
+                                                             partyUInfo:apu
+                                                             partyVInfo:apv
+                                                                  error:&concatKDFError];
+
+    if (!derivedKey)
+    {
+        if (error)
+        {
+            *error = concatKDFError;
+        }
+        
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to calculate ConcatKDF with error %@", concatKDFError);
+        return nil;
+    }
+    
+    return derivedKey;
+}
+
+- (NSDictionary *)decryptJweResponseUsingSymmetricKey:(NSData *)symmetricKey
+                                                error:(NSError * _Nullable __autoreleasing * _Nullable)error
+{
+    if (!symmetricKey)
+    {
+        if (error)
+        {
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Symmetric key is nil", nil, nil, nil, nil, nil, NO);
+        }
+        return nil;
+    }
+    
+    if (![self IsJweResponseAlgorithmSupported:error])
+    {
+        return nil;
+    }
+    
+    // Since only A256GCM is supported, we can decrypt jwe message using AES256GCM.
+    MSIDAesGcmDecryptor *decryptor = [MSIDAesGcmDecryptor new];
+    NSData *decryptedData = [decryptor decryptWithAES256GCMHandlerWithMessage:self.payload iv:self.iv key:symmetricKey tag:self.tag aad:self.aad error:error];
```

Comment:
Does this method print the actual error description thrown from crypto layer so we can debug if decryption not successful?

Reply:
Updated the error function MSIDCreateError to print error description.